src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -137,23 +137,26 @@ public AD7DebugSession(Stream debugAdapterStdIn, Stream debugAdapterStdOut, List
         private void HandleSendInvalidateRequestAsync(IRequestResponder<SendInvalidateArguments> responder)
         {
             InvalidatedEvent invalidated = new InvalidatedEvent();
-            
-            // Setting the area and adding it to the result
-            invalidated.Areas.Add(responder.Arguments.Areas);
+            // Set the Arguments only if passed
+            if (null != responder.Arguments) {
+                // Setting the Areas if passed
+                if (null != responder.Arguments.Areas || responder.Arguments.Areas.Length != 0) {
```

Comment:
```suggestion
                if (null != responder.Arguments.Areas) {
```

It is ok to send an empty array with invalidateEvent.

>  If this property is missing, empty, or if values are not understood, the client should assume a single value `all`.",